molo/core/views.py

```diff
@@ -130,18 +130,22 @@ def add_translation(request, page_id, locale):
     new_title = str(language) + " translation of %s" % page.title
     new_slug = generate_slug(new_title)
     translation = page.__class__(
-        title=new_title, slug=new_slug)
+        title=new_title, slug=new_slug, language=language)
     page.get_parent().add_child(instance=translation)
     translation.save_revision()
-    language_relation = translation.languages.first()
-    language_relation.language = language
-    language_relation.save()
-    translation.save_revision()
+    # add the translation the new way
+    page.specific.translated_pages.add(translation)
```

Comment:
Where do we do the reverse of this? where do we add the main language page to the `translated pages` for the new language page?

Reply:
added it now.